lib/semantics/assignment.h

```diff
@@ -31,12 +32,23 @@ struct ForallStmt;
 struct ForallConstruct;
 }
 
+namespace Fortran::evaluate {
+void CheckPointerAssignment(parser::ContextualMessages &, const Symbol &,
+    const evaluate::Expr<evaluate::SomeType> &);
+}
```

Comment:
It looks like this declaration isn't supposed to be part of this change?

Reply:
I left it in, it's just a prototype.